deafrica/utils.py

```diff
@@ -208,6 +210,19 @@ def test_http_return(returned):
         )
 
 
+def io_timer(file_path: Path, log: Optional[logging.Logger] = None):
+    """
+    Function to check if file exist.
+    """
+    # give extra time to output write the file in disk
+    count = 0
+    while not os.path.exists(file_path) and count < 4:
+        if log:
+            log.info(f"Wait OS write file in disk {count}s")
+        time.sleep(1)
+        count += 1
```

Comment:
I don't get this

Reply:
@NikitaGandhi and I found out that the main issue in Argo was that when the second task tried to get the file, the file wasn't in the folder yet. It means that Argo was faster to execute the lines than write the file on disk.
The way to solve that was by adding a sleep(1.5), which @NikitaGandhi asked me to add a file check instead a sleep, so I added this function which checks the file and if not there sleeps a bit and checks again. I added a limit to never be in an infinit loop.

Reply:
I totally agree with you, and I was totally against this solution when Nikita asked me to try, however, we tried and worked ..... We couldn't find any other explanation.
As always I'm open to other and better options.

Reply:
PS=> Locally running in my Docker, the process worked just fine without the use of this function.